fix: issue with docker config and volume setup

Extheoisah opened the pull request:
- modifies the volume setup script for sim.json config
- adds --data-dir to CMD args (default is /data in volume)
- fix entrypoint.sh script

carlaKC commented on docker/setup-volume.sh:
```diff
@@ -28,11 +28,10 @@ NODE_COUNT=$(cat $SIMFILE_PATH_ON_HOST | jq '.nodes | length')
-    NODE_TLS_PATH_ON_HOST=$(cat $SIMFILE_PATH_ON_HOST | jq -r ".nodes[$i].$NODE_TYPE.cert") # TLS path
-    NODE_MACAROON_PATH_ON_HOST=$(cat $SIMFILE_PATH_ON_HOST | jq -r ".nodes[$i].$NODE_TYPE.macaroon") # Macaroon path
-    
+    NODE_ID=$(cat $SIMFILE_PATH_ON_HOST | jq -r ".nodes[$i].id") # Extract node ID for directory creation.
```
commit structure nitpick: in the commit we're fixuping up two issues: 
* Change to config structure made in #152 
* Updated to path specification in #132 

Ideally we'd break this up into two commits, so that each contains one "logical idea". Makes it easier to review and better to bisect/debug in future if we need to :)

Extheoisah replied:
fixed. created two new commits.

carlaKC commented on Makefile:
```diff
@@ -21,7 +21,7 @@ help:
 	@echo "PRINT_BATCH_SIZE  Set the batch size for printing the results e.g. <make run PRINT_BATCH_SIZE=100>"
 	@echo "TOTAL_TIME        Set the total time for the simulation e.g. <make run TOTAL_TIME=1000>"
-	@echo "DATA_DIR		  	 Set the data directory for the simulation containing simulation files and results e.g. <make run DATA_DIR=/data>"
+	@echo "DATA_DIR		  	 Set the data directory for the simulation containing simulation files and results e.g. <make run-docker DATA_DIR="/Users/anon/data>""
```
Spacing looks off here?

Extheoisah replied:
hmm.... spacing looks fine on local.

carlaKC replied:
Looks like you're using tabs and the rest are spaces - can manually replace with spaces or set your editor to convert tabs to spaces.

Extheoisah replied:
fixed.

carlaKC reviewed (commented):
Looks good, just some structural comments which I'd like to fix up since we need to wait for #132 anyway.

carlaKC reviewed (approved):
verynice! 

I'll test this out locally once #132 is done, then merge 🚀

sr-gi commented on Makefile:
```diff
@@ -15,14 +16,15 @@ help:
 	@echo ""
 	@echo "Variables:"
-	@echo "SIMFILE_PATH       Path to the sim.json file."
+	@echo "SIMFILE_PATH      Path to the sim.json file."
```
nit: this line and the one for `stop` three lines above end with a `.`, while the rest do not.

sr-gi commented on Makefile:
```diff
@@ -15,14 +16,15 @@ help:
-	docker run -d --rm --name sim-ln --init -v simln-data:/data -e SIMFILE_PATH=/data/sim.json -e LOG_LEVEL=$(LOG_LEVEL) -e HELP=${HELP} -e PRINT_BATCH_SIZE=${PRINT_BATCH_SIZE} -e TOTAL_TIME=${TOTAL_TIME} sim-ln
+	docker run -d --rm --name sim-ln --init -v simln-data:/data -e SIMFILE_PATH=/data/sim.json -e DATA_DIR=${DATA_DIR} -e LOG_LEVEL=$(LOG_LEVEL) -e HELP=${HELP} -e PRINT_BATCH_SIZE=${PRINT_BATCH_SIZE} -e TOTAL_TIME=${TOTAL_TIME} sim-ln
 
 run-interactive:
```
I didn't catch this when it was added, but I'm wondering now:

Don't we want to also pass the env variables when running interactively?

Extheoisah replied:
hmm... we should. I'll fix that.

Extheoisah replied:
fixed here -> https://github.com/bitcoin-dev-project/sim-ln/pull/159/commits/85082b3b80ac1299cf750a239544594b2ac2df3e

sr-gi commented on docker/README.md:
```diff
@@ -81,6 +81,7 @@ Other configurable variables include:
 - `HELP`: Set to `true` to print the help message.
 - `PRINT_BATCH_SIZE`: determines the number of payment results that will be written to disk at a time.
 - `TOTAL_TIME`: the total runtime for the simulation expressed in seconds.
+- `DATA_DIR`: Path to a directory containing simulation files, and where simulation results will be stored (default is the `/data` volume directory)
```
nit: missing trailing `.`

sr-gi commented on docker/entrypoint.sh:
```diff
@@ -1,7 +1,12 @@
 
 # Define the start command
-START_COMMAND="/usr/local/bin/sim-cli $SIMFILE_PATH"
+START_COMMAND="/usr/local/bin/sim-cli --sim-file $SIMFILE_PATH"
```
Shouldn't this only be passed now if `$SIMFILE_PATH` is defined? If not passed, this will default to pick it from `data_dir/sim.json`

Extheoisah replied:
We still need it because it looks for the config JSON in the root by default. whereas in docker, we're mounting a volume that hosts the JSON. So we need to point it to the docker volume where the sim.json is hosted and use that as a default if no SIMFILE_PATH is specified

sr-gi commented on Makefile:
```diff
@@ -15,14 +16,15 @@ help:
 
 run-docker:
-	docker run -d --rm --name sim-ln --init -v simln-data:/data -e SIMFILE_PATH=/data/sim.json -e LOG_LEVEL=$(LOG_LEVEL) -e HELP=${HELP} -e PRINT_BATCH_SIZE=${PRINT_BATCH_SIZE} -e TOTAL_TIME=${TOTAL_TIME} sim-ln
+	docker run -d --rm --name sim-ln --init -v simln-data:/data -e SIMFILE_PATH=/data/sim.json -e DATA_DIR=${DATA_DIR} -e LOG_LEVEL=$(LOG_LEVEL) -e HELP=${HELP} -e PRINT_BATCH_SIZE=${PRINT_BATCH_SIZE} -e TOTAL_TIME=${TOTAL_TIME} sim-ln
```
`-e SIMFILE_PATH=/data/sim.json` should be `-e SIMFILE_PATH=${SIMFILE_PATH}` now right? Given the parameter is not required anymore plus we want the user to be able to re-define the location if desired

Extheoisah replied:
Gotcha. fixed.

sr-gi reviewed (commented):
Looks good to me overall.

Some comments and nits inline.

m3dwards commented:
Looks straightforward enough.

utACK 85082b3b80ac1299cf750a239544594b2ac2df3e

carlaKC reviewed (commented):
Came back to test and ran into issues with `make docker build` and `make run-interactive`, I think we've made some breaking changes since you last updated :')

```
docker run -it --rm --name sim-ln --init -v simln-data:/data_dir -e SIMFILE_PATH=/data_dir/sim.json -e DATA_DIR= -e LOG_LEVEL=info -e HELP= -e PRINT_BATCH_SIZE= -e TOTAL_TIME= sim-ln
error: unexpected argument '--sim-file' found

  tip: to pass '--sim-file' as a value, use '-- --sim-file'
```

Extheoisah commented:
> Came back to test and ran into issues with `make docker build` and `make run-interactive`, I think we've made some breaking changes since you last updated :')
> 
> ```
> docker run -it --rm --name sim-ln --init -v simln-data:/data_dir -e SIMFILE_PATH=/data_dir/sim.json -e DATA_DIR= -e LOG_LEVEL=info -e HELP= -e PRINT_BATCH_SIZE= -e TOTAL_TIME= sim-ln
> error: unexpected argument '--sim-file' found
> 
>   tip: to pass '--sim-file' as a value, use '-- --sim-file'
> ```

Hmm...maybe. I haven't run it since. I'll look into it

Extheoisah commented on Makefile:
```diff
@@ -15,17 +17,18 @@ help:
+	@echo "DATA_DIR          Set the data directory for the simulation containing simulation files and results e.g. <make run-docker DATA_DIR="/Users/anon/data_dir>"."
 
 run-docker:
-	docker run -d --rm --name sim-ln --init -v simln-data:/data -e SIMFILE_PATH=/data/sim.json -e LOG_LEVEL=$(LOG_LEVEL) -e HELP=${HELP} -e PRINT_BATCH_SIZE=${PRINT_BATCH_SIZE} -e TOTAL_TIME=${TOTAL_TIME} sim-ln
```
Ah! I will fix this.

Extheoisah replied:
fixed here -> https://github.com/bitcoin-dev-project/sim-ln/pull/159/commits/bbb168da101ecd7f79977cee08389a54f54ae459

Extheoisah commented:
> Came back to test and ran into issues with `make docker build` and `make run-interactive`, I think we've made some breaking changes since you last updated :')
> 
> ```
> docker run -it --rm --name sim-ln --init -v simln-data:/data_dir -e SIMFILE_PATH=/data_dir/sim.json -e DATA_DIR= -e LOG_LEVEL=info -e HELP= -e PRINT_BATCH_SIZE= -e TOTAL_TIME= sim-ln
> error: unexpected argument '--sim-file' found
> 
>   tip: to pass '--sim-file' as a value, use '-- --sim-file'
> ```

@carlaKC So I tried running locally with the new changes and the only error I had was due to limitations in my docker disk size (not related to sim-ln) for which I added a small clean-up here -> https://github.com/bitcoin-dev-project/sim-ln/pull/159/commits/bbda19fe0e038ce97418e688a2037bd5dd17d823
I cleaned up previous sim-ln build images and volumes and created new ones and successfully ran `make build docker` and `make run-interactive`. Please clean up your docker setup and try again. If the issue continues, could you send a screenshot of the error?

carlaKC reviewed (approved):
tACK, I think I had some caching issues on my end. Thanks for the update!
